### ISSUE TITLE
Google is not showing AMO results of uBlock Origin on the first page

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-frontend#7623]
This issue is related to [this Reddit thread](https://www.reddit.com/r/firefox/comments/87zsml/google_is_not_showing_amo_results_of_ublock/) (as well as some random discussions lost somewhere in one of our communication tools). Anyway, depending on which Google TLD is used, which preferred language is set and probably other unknown parameters, the AMO link to "ublock origin" does not always appear in the search results (in the first page(s)). **This does not happen with other popular add-ons, though.**

The Reddit thread is rather old, it was posted 11 months ago. AFAIK we (Mozilla) also noticed the problem, but only a few months ago (~ December 2018) if I am not mistaken. The first thought that came to some of us was that we made [some SEO improvements in Q4 2018](https://github.com/orgs/mozilla/projects/35).

According to the Reddit thread, it is much older so the SEO improvements did not improve the situation but they did not necessarily impact the performances of the query "ublock origin" in Google either.

Performances of this query over the last six months (retrieved on Feb 21st, 2019):

![](https://irccloud.mozilla.com/file/HdAxM11f/Screen%20Shot%202019-02-21%20at%2016.22.40.png)

At first glance, something happened in Q4 2018. I am not entirely sure but the Googlebot has troubles retrieving the mobile version of AMO, but not always of course.. The old Google "console" successfully fetches and renders the ublock origin detail page but the new Google "search tool" fails, sometimes.

Some of the errors are related to [the code-splitting feature](https://github.com/mozilla/addons-frontend/issues/3734) that indeed landed in Q4 2018. I can't prove that this is the reason why the query "ublock origin" does not perform well in Google, but I can't prove that it is not the cause either, except that other popular queries (i.e. popular add-on names) perform much better.

That being said, if we fix the issues reported by Googlebot and related to the code-splitting, which seems to only be a problem for the Googlebot (and it is not entirely clear why to me yet), I suppose it could positively impact the overall performances of different search queries, and maybe it could even do something for the "ublock origin" query.

There is no guarantee though, but that's still something worth doing in my opinion.

**Comment by jvillalobos** [comment]
@rraue was looking into this as well. Also, @atsay had found other extensions that had this same problem.

**Comment by muffinresearch** [comment]
Downgrading this to a P3 until we hear from @rraue with any pointers towards a fix.

**Comment by willdurand** [comment]
Google reports the following coverage data for "uBlock Origin" when inspecting: https://addons.mozilla.org/en-US/firefox/addon/ublock-origin/

```
Referring page: https://addons.mozilla.org/firefox/addon/ublock-origin/
```

If I try to inspect: https://addons.mozilla.org/en-US/firefox/addon/privacy-badger17/, we get:

```
Referring page: https://addons.mozilla.org/en-US/firefox/addon/privacy-badger17
```

This sounds better as the `locale` is part of this referring page. It might be totally unrelated but that's one difference between uBlock Origin and PB.

**Comment by willdurand** [comment]
Other add-ons with the similar issue:

- https://addons.mozilla.org/en-US/firefox/addon/grammarly-1/
- https://addons.mozilla.org/en-US/firefox/addon/duckduckgo-for-firefox/
- https://addons.mozilla.org/en-US/firefox/addon/1password-x-password-manager/
- https://addons.mozilla.org/en-US/firefox/addon/amazon-browser-bar/
- https://addons.mozilla.org/en-US/firefox/addon/enhancer-for-youtube/

---

### Referring pages

For Grammarly:

```
https://addons.mozilla.org/en-US/firefox/addon/grammarly-1
```

For DuDuckGo:

```
https://addons.mozilla.org/firefox/addon/duckduckgo-for-firefox/
```

For 1Password:

```
https://addons.mozilla.org/firefox/addon/1password-x-password-manager/
```

For Amazon Assistant:

```
None detected
```

For Youtube:

```
https://addons.mozilla.org/en-us/firefox/addon/enhancer-for-youtube/
```

**Comment by willdurand** [comment]
That invalidates the theory in my previous comment but it could still be related, "most" referring pages are weird for those add-ons.

**Comment by willdurand** [comment]
This issue is now fixed, most likely by #9309
